src/main/javascript/crypto/src/eas/EASIdentifierAttestation.ts

```diff
@@ -0,0 +1,54 @@
+import {EasTicketAttestation, EasTicketCreationOptions} from "./EasTicketAttestation";
+import {Signer} from "ethers";
+import {KeyPair, KeysArray} from "../libs/KeyPair";
+
+export interface EasIdentiferParameters {
+	version: number;
+	identifierType: "email";
+	commitment: string;
+	ethereumAddress: string;
+}
+
+export class EASIdentifierAttestation extends EasTicketAttestation {
+
+	constructor(
+		signingConfig?: {
+			EASconfig: {
+				address: string // EAS resolver contract address
+				version: string
+				chainId: number
+			},
+			signer: Signer
+		},
+		issuerKey?: KeyPair|string
+	) {
+		const schema = {
+			fields: [
+				{name: "version", type: "uint8"},
+				{name: "identifierType", type: "string"},
+				{name: "commitment", type: "bytes", isCommitment: true},
+				{name: "ethereumAddress", type: "address"}
+			]
+		};
+
+		super(schema, signingConfig, undefined, issuerKey ? <KeysArray>{"": issuerKey} : undefined);
+	}
+
+	async createEasAttestation(data: EasIdentiferParameters|any, options?: EasTicketCreationOptions, commitmentType = 'mail'){
+
+		if (!options)
+			options = {};
+
+		// Identifier attestation is not revocable at this stage.
+		if (options.revocable === undefined)
+			options.revocable = false;
+
+		if (!options.validity){
+			const date = new Date();
+			date.setDate(date.getDate() + 30);
```

Comment:
better add input variable for TTL of attestation  in seconds so user developer can easy change it.
@micwallace

Reply:
@oleggrib validity from and to date. This can be easily changed by the developer by passing these options. By default, for identifier attestations the validity is 30 days as per the current ASN implementation.